spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala

```diff
@@ -22,11 +22,36 @@ import com.intel.analytics.bigdl.utils.Table
 
 import scala.reflect.ClassTag
 
+/**
+ * A plain implementation of SGD
+ * RETURN:
+ * - x     : the new x vector
+ * - f(x)  : the function, evaluated before the update
+ * @tparam T data type
+ */
 class SGD[@specialized(Float, Double) T: ClassTag](implicit ev: TensorNumeric[T])
   extends OptimMethod[T] {
 
   import SGD._
 
+  /**
+   *
+   * @param feval  a function that takes a single input (X), the point of a evaluation,
+   *               and returns f(X) and df/dX
+   * @param x      the initial point
+   * @param config a table with configuration parameters for the optimizer
+   *               learningRate      : learning rate
```

Comment:
please don't align the parameters.
We can't guarantee this alignment on all environments.